File: LogicadeProgramacao/Estrutura de Decisão/3. Python/NumCentenasDezenasUnidades.py
numero = int(input("Digite um número inteiro menor que 1000: "))

centenas = numero // 100
resto = numero % 100
dezenas = resto // 10
unidades = resto % 10

if centenas > 0:
    print(f"{centenas} centena(s)", end="")
    if dezenas > 0 and unidades > 0:
        print(", ", end="")
    elif dezenas > 0 or unidades > 0:
        print(" e ", end="")
    
if dezenas > 0:
    print(f"{dezenas} dezena(s)", end="")
    if unidades > 0:
        print(" e ", end="")
    
if unidades > 0:
    print(f"{unidades} unidade(s)", end="")

File: LogicadeProgramacao/Estrutura de Decisão/2. Java/NumCentenasDezenasUnidades.java
import java.util.Scanner;

public class NumCentenasDezenasUnidades {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        System.out.print("Digite um número inteiro menor que 1000: ");
        int numero = scanner.nextInt();
        
        int centenas = numero / 100;
        int resto = numero % 100;
        int dezenas = resto / 10;
        int unidades = resto % 10;
        
        if (centenas > 0) {
            System.out.print(centenas + " centena(s)");
            if (dezenas > 0 && unidades > 0) {
                System.out.print(", ");
            } else if (dezenas > 0 || unidades > 0) {
                System.out.print(" e ");
            }
        }
        
        if (dezenas > 0) {
            System.out.print(dezenas + " dezena(s)");
            if (unidades > 0) {
                System.out.print(" e ");
            }
        }
        
        if (unidades > 0) {
            System.out.print(unidades + " unidade(s)");
        }
    }
}
